clang/tools/driver/cc1depscan_main.cpp

```diff
@@ -1034,7 +1032,7 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
 
   // Wait for the work to finish.
   const uint64_t SecondsBetweenAttempts = 5;
-  const uint64_t SecondsBeforeDestruction = 15;
+  const uint64_t SecondsBeforeDestruction = LongRunning ? 45 : 15;
```

Comment:
Maybe the better option is to specify a timeout to daemon instead of just saying `longrunning`?

Reply:
That'd probably be useful if we want to allow the user or the build system to control it, I'm currently not convinced we need to expose it further but later we can re-evaluate if there's a compelling reason.